lightning/src/ln/channel.rs

```diff
@@ -4812,11 +4837,21 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<&'a K> for Channel<Signer>
 		let counterparty_dust_limit_satoshis = Readable::read(reader)?;
 		let holder_dust_limit_satoshis = Readable::read(reader)?;
 		let counterparty_max_htlc_value_in_flight_msat = Readable::read(reader)?;
-		let counterparty_selected_channel_reserve_satoshis = Readable::read(reader)?;
+		let mut counterparty_selected_channel_reserve_satoshis = Some(Readable::read(reader)?);
+		if counterparty_selected_channel_reserve_satoshis == Some(0) {
+			// Versions up to 0.0.98 had counterparty_selected_channel_reserve_satoshis as a
+			// non-option, writing 0 for what we now consider None.
+			counterparty_selected_channel_reserve_satoshis = None;
+		}
 		let counterparty_htlc_minimum_msat = Readable::read(reader)?;
 		let holder_htlc_minimum_msat = Readable::read(reader)?;
 		let counterparty_max_accepted_htlcs = Readable::read(reader)?;
-		let minimum_depth = Readable::read(reader)?;
+		let mut minimum_depth = Some(Readable::read(reader)?);
+		if minimum_depth == Some(0) {
+			// Versions up to 0.0.98 had minimum_depth as a non-option, writing 0 for what we now
+			// consider None.
+			minimum_depth = None;
+		}
```

Comment:
Thinking out loud: unsure if these special cases don't bode well for the complexity of our backwards compat going forward :thinking:

Reply:
Right, I'm not sure how to do it particularly better - its nice to be backwards compatible while still being able to add new things. At least in this case if we ever need to write 0s we can just bump the version number and then ignore this value when reading a new channel. I figured just shoving the "real" value in the TLV is nice and simple, but we could do this another way if we prefer.

Reply:
If we get review on #975, we could use the version bump included there to just decide whether to read the value here or in the TLV (in the same way we do there) to also change the read here. That way we'd avoid two version bumps as well, which is nice.

Reply:
I see! I don't really mind, up to you

Reply:
I added a commit from #975 to do this, makes sense if we merge for 0.0.99.